### ISSUE TITLE
[Assert] Assigning an image asset in the Image Gradient component results in an assert

### ISSUE THREAD

**Issue by LB-JakubSkorupka** [o3de/o3de#7571]
**Describe the bug**
Assigning **any** image asset in the **Image Gradient** component results in assert being printed to the console, the assert condition differs depending on image asset used and its **DirectX Pixel Format**. 
Three unsupported **DirectX Pixel Formats** have been found, we were not able to confirm if any other **DirectX Pixel Formats** are affected by this issue.

Please note that each assert mentions the exact same file and line:

**File**: `C:/builds/o3de/Gems/Atom/RPI/Code/Source/RPI.Public/RPIUtils.cpp`

**Line**: `168`

For more information, please refer to the attached **Editor.log** files.

The following asserts are being printed to the console:

**Assert 1**

**Assigned file**: legacylogolauncher.bmp

**Condition**
```
(4360) 'float __cdecl AZ::RPI::Internal::RetrieveFloatValue(const unsigned char *,unsigned __int64,enum AZ::RHI::Format)'
(System) - Unsupported pixel format: BC1_UNORM
```

**Assert 2**

**Assigned file**: preview.png

**Condition**
```
(5544) 'float __cdecl AZ::RPI::Internal::RetrieveFloatValue(const unsigned char *,unsigned __int64,enum AZ::RHI::Format)'
(System) - Unsupported pixel format: BC3_UNORM_SRGB
```

**Assert 3**

**Assigned file**: default_level_cubemap_iblspecular.tif

**Condition**
```
(14412) 'float __cdecl AZ::RPI::Internal::RetrieveFloatValue(const unsigned char *,unsigned __int64,enum AZ::RHI::Format)'
(System) - Unsupported pixel format: R9G9B9E5_SHAREDEXP
```

**Steps to reproduce**
1. Create a new entity.
2. Add an **Image Gradient** component to the created entity.
3. Add components that are required by **Image Gradient** component. (Gradient Modifier, Shape component)
4. Assign any image asset in the **Image Gradient** component.

**Expected behavior**
The asset is assigned without issues in the **Image Gradient** component.

**Actual behavior**
An assert is printed to the console when assigning the image asset in the **Image Gradient** component.

**Video**

https://user-images.githubusercontent.com/86951967/153579016-80c87519-4013-46ea-9707-940d636ad2cb.mp4

**Found in Branch**
Development (257f3cb)

**Desktop:**
 - Device: PC
 - OS: Windows
 - Version: 10
 - CPU: Intel Core i7-8700K
 - GPU: NVIDIA GeForce RTX 2080 Ti
 - Memory: 16GB

**Additional context**
[Editor.log](https://github.com/o3de/o3de/files/8047992/Editor.log)
[Editor.log](https://github.com/o3de/o3de/files/8047995/Editor.log)
[Editor.log](https://github.com/o3de/o3de/files/8047999/Editor.log)

**Comment by cgalvan** [comment]
With https://github.com/o3de/o3de/pull/7590 the user can no longer assign an image with an unsupported format for an Image Gradient Component image asset.

**Comment by LB-JakubSkorupka** [comment]
Issue verified as fixed on build Development (3f2aa74).
